### ISSUE TITLE
Review my submission

### ISSUE THREAD

**Issue by sukhada** [sukhada/Flickster#1]
Project 1 - Flickster

Flickster shows the latest movies currently playing in theaters. The app utilizes the Movie Database API to display images and basic information about these movies to the user.

Time spent: 11 hours spent in total

User Stories

The following required functionality is completed:
- [x]  User can scroll through current movies from the Movie Database API
- [x]  Layout is optimized with the ViewHolder pattern.
  For each movie displayed, user can see the following details:
- [x]  Title, Poster Image, Overview (Portrait mode)
- [x]  Title, Backdrop Image, Overview (Landscape mode)
  The following optional features are implemented:
- [x]  User can pull-to-refresh popular stream to get the latest movies.
- [x]  Display a nice default placeholder graphic for each image during loading.
- [x]  Improved the user interface through styling and coloring.
  The following bonus features are implemented:
- [x]  Allow user to view details of the movie including ratings and popularity within a separate activity or dialog fragment.
- [ ] When viewing a popular movie (i.e. a movie voted for more than 5 stars) the video should show the full backdrop image as the layout. Uses Heterogenous ListViews or Heterogenous RecyclerView to show different layouts.
- [x]  Allow video trailers to be played in full-screen using the YouTubePlayerView.
- [ ] Overlay a play icon for videos that can be played.
- [ ] More popular movies should start a separate activity that plays the video immediately.
- [ ] Less popular videos rely on the detail page should show ratings and a YouTube preview.
- [ ] Leverages the data binding support module to bind data into layout templates.
- [ ] Apply the popular Butterknife annotation library to reduce boilerplate code.
- [ ] Apply rounded corners for the poster or background images using Picasso transformations

The following additional features are implemented:
- [ ] List anything else that you can get done to improve the app functionality!

I added a RecyclerView pattern and am prefetching images so that images take less time to load/appear on the screen. 

![flickster](https://cloud.githubusercontent.com/assets/1696078/17469565/4a42cfe8-5ce9-11e6-9286-6c1de1f8182c.gif)

**Comment by rogerhu** [comment]
Can you use the submit button on the course portal and include an animated gif?

**Comment by sukhada** [comment]
@rogerhu yep, added a GIF with it. I was testing with a physical phone, so the GIF doesn't include the Youtube Player stuff, since I didn't install the Play Services APK for my emulator. :(

**Comment by lita** [comment]
## Week 1: Flicks

👍 good work, shashank! A few notes after checking out the code:
- Please remember to add a README.md for every project! Not just adding it to the issue.
- See feedback guide below for common conventions and standards around naming, using string resources, etc
- Consider including a check to see if the internet is available to detect any network failures. Show an appropriate message in case of failure.
- It is a bit weird to see your activity file in the `models` folder. consider adding an `activity` folder as well.
- Good to see you playing around with RelativeLayout. RelativeLayout is one of the most powerful responsive-first layout systems available across web and mobile platforms and provides better control in comparison to a linear layout.
- Nice work handling  `onFailure` method in `JsonHttpResponseHandler`. Consider adding an error dialog here for the user when a network request fails.
- Nice to see you used the SwipeRefreshLayout for easy stream refreshes
- Great work on using a [placeholder graphic](http://guides.codepath.com/android/Displaying-Images-with-the-Picasso-Library#configuring-picasso) during loading.
- Great work on using the [YouTubePlayerView](http://guides.codepath.com/android/Streaming-Youtube-Videos-with-YouTubePlayerView).
- Consider using the [ButterKnife library annotations](http://guides.codepath.com/android/Reducing-View-Boilerplate-with-Butterknife) for declaring handles to view objects.

One of the most important part of these projects is that you add additional features and tweak the UI / UX because that will provide many more learning opportunities. I would encourage you to complete the projects each week with required stories early and then spend time adding your own UI elements and experimenting with optional extensions that will improve the user experience.

We have provided a detailed [Project 1 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_1_feedback) which covers the most common points we see for this project. Read through the [feedback guide point-by-point](http://courses.codepath.com/snippets/intro_to_android/project_1_feedback) to determine other ways you could improve your submission. You should consider going back and implementing applicable feedback as well. Keep in mind that one of the most important parts of Android development is learning the correct patterns and conventions.

Hopefully, this first project has given you a better sense of working with RelativeLayout which is a very flexible layout system, probably one of the most powerful responsive-first layout systems available across web and mobile platforms. This assignment also gave us our first introduction to networking, working with APIs and loading remote images. The next assignment will introduce new concept such as accepting user input and navigating between activities but will also reinforce important concepts such as networking, using APIs, handling remote images.

If you have any particular questions about the assignment in general or on any of the feedback, feel free to reply here or slack me at @lita in the Hearsay channel. 

**P.S.** Good to see you properly added the README with features and screenshots to your project as described in the [submitting assignments](http://courses.codepath.com/courses/intro_to_android/pages/submitting_assignments) guide! Hopefully, the submit button worked out. Let me know if you ran into any issues.
